src/frontend/apps/standalone_site/src/features/PortabilityRequests/components/ItemTableRow.tsx

```diff
@@ -0,0 +1,133 @@
+import { Box, Button } from 'grommet';
+import { FormCheckmark, FormClose } from 'grommet-icons';
+import { PortabilityRequest, Spinner } from 'lib-components';
+import React from 'react';
+import { defineMessages, useIntl } from 'react-intl';
+
+import {
+  acceptPortabilityRequest,
+  rejectPortabilityRequest,
+} from '../api/usePortabilityRequests';
+
+import { PortabilityRequestStateTag } from './PortabilityRequestStateTag';
+
+const messages = defineMessages({
+  acceptPortabilityRequest: {
+    defaultMessage: 'Accept',
+    description: 'Accessibility text for portability request accept button',
+    id: 'features.PortabilityRequests.ItemTableRow.acceptPortabilityRequest',
+  },
+  rejectPortabilityRequest: {
+    defaultMessage: 'Reject',
+    description: 'Accessibility text for portability request reject button',
+    id: 'features.PortabilityRequests.ItemTableRow.rejectPortabilityRequest',
+  },
+  requestFromLtiUser: {
+    defaultMessage: 'Made by LTI user',
+    description:
+      'Value displayed in the table when the request is from an LTI user',
+    id: 'features.PortabilityRequests.ItemTableRow.requestFromLtiUser',
+  },
+  rowPortabilityRequestText: {
+    defaultMessage: '{from_playlist} wants access to {for_playlist}',
+    description:
+      'Row text displayed in the table to display the playlists involved in the request',
+    id: 'features.PortabilityRequests.ItemTableRow.rowPlaylistText',
+  },
+});
+
+interface ItemTableRowProps {
+  item: PortabilityRequest;
+}
+
+const AcceptRejectButtons = ({ item }: ItemTableRowProps) => {
+  const intl = useIntl();
+  const [displayButtons, setDisplayButtons] = React.useState(item.allow_action);
+  const currentItemId = React.useRef(item.id);
```

Comment:
Why do you need a useRef here btw?

Reply:
Because its value is updated in the useEffect 
```
  React.useEffect(() => {
    if (currentItemId !== item.id) {
      currentItemId = item.id;
      setDisplayButtons(item.allow_action);
    }
  }, [item.id, item.allow_action]);
```
I'm interested in a rewrite of this if it could be simpler :) 

My initial problem was :

- When the item is accepted/rejected the buttons are hidden (expected)
- When the component is re-rendered for the same item it's okay, the buttons stays hidden
- When the component is re-rendered for another item, I need to somehow display/hide the buttons according to the new item. 

I tried using directly the `item.allow_action` attribute but it does not update when I try a rerender in `accepts portability request` test.

Reply:
Yes ok. 
To be able to be simpler and remove this useEffect, the row "key" in SortableTable `key=row_${index}` should have been unique based on the item and not the index, ex: `key=row_${item.id}`. 
But `item` is generic, we don't know his properties, it is why it is like that I guess.